src/transformers/models/beit/modeling_beit.py

```diff
@@ -780,20 +778,42 @@ def forward(
         )
 
         sequence_output = outputs[0]
-        sequence_output = self.layernorm(sequence_output)
-        prediction_scores = self.lm_head(sequence_output[:, 1:])
 
-        masked_lm_loss = None
-        if labels is not None:
-            loss_fct = CrossEntropyLoss()  # -100 index = padding token
-            masked_lm_loss = loss_fct(prediction_scores[bool_masked_pos], labels)
+        masked_im_loss = None
+        if self.config.decoder_type == "beit":
+            sequence_output = self.layernorm(sequence_output)
+            prediction_scores = self.lm_head(sequence_output[:, 1:])
+
+            if labels is not None:
+                loss_fct = CrossEntropyLoss()  # -100 index = padding token
+                masked_im_loss = loss_fct(prediction_scores[bool_masked_pos], labels)
+        elif self.config.decoder_type == "simmim":
+            # Reshape to (batch_size, num_channels, height, width)
+            sequence_output = sequence_output[:, 1:]
+            B, L, C = sequence_output.shape
+            H = W = int(L**0.5)
+            sequence_output = sequence_output.permute(0, 2, 1).reshape(B, C, H, W)
```

Comment:
B, L, C, H, W are not classes, so should not be capitalized, and should not be one-letter variable names.